vmm/src/memory_manager.rs

```diff
@@ -450,14 +503,34 @@ impl MemoryManager {
         Ok(slot)
     }
 
-    pub fn resize(&mut self, desired_ram: u64) -> Result<bool, Error> {
-        if desired_ram > self.current_ram {
-            self.hotplug_ram_region((desired_ram - self.current_ram) as usize)?;
-            self.current_ram = desired_ram;
-            Ok(true)
+    pub fn virtiomem_resize(&mut self, size: u64) -> Result<(), Error> {
+        let region = self.virtiomem_region.take();
+        if let Some(region) = region {
+            self.add_region(region)?;
+        }
+
+        if let Some(resize) = &self.virtiomem_resize {
+            resize.work(size).map_err(Error::VirtioMemResizeFail)?;
         } else {
-            Ok(false)
+            panic!("should not fail here");
+        }
+
+        Ok(())
+    }
+
+    pub fn resize(&mut self, desired_ram: u64) -> Result<bool, Error> {
+        let mut ret = false;
+        if desired_ram >= self.current_ram {
+            let size = desired_ram - self.current_ram;
+            if self.virtiomem_resize.is_some() {
+                self.virtiomem_resize(size)?;
```

Comment:
Also, why don't we set `self.current_ram = desired_ram;` for the virtio-mem path? Is that intentional?

Reply:
In this part, code need knows if desired_ram is bigger than normal memory size.
When resize get desired_ram that is bigger than normal memory size, it will let self.virtiomem_resize handle it.  If not, it will drop it.

I use "self.current_ram" save the normal memory size because just fn resize use it.

Reply:
Added self.boot_ram to make the code clear.